Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs

```diff
@@ -929,7 +929,54 @@ private void DestroyAllServices()
         {
             if (!ExecuteOnAllServices(service => service.Destroy())) { return; }
```

Comment:
>if (!ExecuteOnAllServices(service => service.Destroy())) { return; } [](start = 12, length = 68)

Should services be unregistered before they are destroyed?

It seems like otherwise there's still technically some time during which services are registered but still are running.

Then again, if the intent is that destroy should happen first because this is actually just a notification to the service that it's being shutdown, then that's fine.

The order here seems important but there's no comments/docs about the mechanics, so I figured it would be good to ask.

Reply:
developers can call unregister explicitly if destroy is not desired. this is in the service locator shutdown path.

good question on the ordering. we can move the destroy call to the end of the method for safety

Reply:
moved destroy call to end of method